rllib/utils/replay_buffers/episode_replay_buffer.py

```diff
@@ -518,78 +525,82 @@ def _sample_episodes(
             if random_n_step:
                 actual_n_step = int(self.rng.integers(n_step[0], n_step[1]))
 
+            lookback = int(episode_ts != 0)
             # Skip, if we are too far to the end and `episode_ts` + n_step would go
             # beyond the episode's end.
-            if episode_ts + actual_n_step > len(episode):
+            if episode_ts + actual_n_step + batch_length_T + lookback > len(episode):
                 continue
 
+            sampled_episode = episode.slice(
+                slice(
+                    episode_ts - lookback,
+                    episode_ts + actual_n_step + batch_length_T - 1,
+                ),
+                len_lookback_buffer=lookback,
```

Comment:
This arg should already take care of the slice-adjustment, so you don't need to do this math above anymore:
```
sampled_episode = episode.slice(
    slice(episode_ts, episode_ts + batch_length_T + (actual_n_step - 1)),
    len_lookback_buffer=lookback,
)
```